typescript/packages/toolshed/routes/ai/spell/spell.handlers.ts

```diff
@@ -275,3 +311,28 @@ Respond with ${
     many ? "an array of valid JSON objects" : "a single valid JSON object"
   }.`;
 }
+
+export const search: AppRouteHandler<SearchSchemaRoute> = async (c) => {
+  const redis = c.get("blobbyRedis");
+  if (!redis) throw new Error("Redis client not found in context");
+  const logger = c.get("logger");
+
+  const startTime = performance.now();
+  const body = (await c.req.json()) as SearchSchemaRequest;
+
+  try {
+    logger.info({ query: body.query }, "Processing search request");
+
+    const result = await performSearch(body.query, logger, redis);
+
+    const response = result;
```

Comment:
The HTTP layer becomes super simple here, all the work is lifted into the behavior tree once the dependencies are provided.